Apps/Common/src/AccessManagement/Authorization/Handlers/FhirResourceAuthorizationHandler.cs

```diff
@@ -219,5 +240,27 @@ private bool ValidateObservationDelegate(AuthorizationHandlerContext context, st
 
             return retVal;
         }
+
+        private bool IsDelegateExpired(string resourceHDID)
+        {
+            if (!this.maxDependentAge.HasValue)
+            {
+                this.logger.LogInformation($"Delegate expired check on resource {resourceHDID} skipped as max dependent age is null");
+                return false;
+            }
+
+            if (this.patientService == null)
+            {
+                this.logger.LogError($"Delegate expired check on resource {resourceHDID} failed as patientService is null");
```

Comment:
I think this should throw since it should never reach this place if going though the correct flow

Reply:
It shouldn't throw but you shouldn't have to do the check... Why are the delegates nullable?  Probably an easy fix when we first introduced the nullable checks but we should change them to non-nullable now and remove the check.

Reply:
+1

Reply:
If this is not nullable, any project that uses the resource authorization handler would have to initialize patient services in their startup which i think its all projects at this point, but only the lab will actually reach this code that checks for delegation. I think theres no harm so far making it not nullable now since the projects won't reach this code

Reply:
i just think its better to get this error instead of a nullreference exception

Reply:
I just tested and i was wrong, all projects would have to initialize patient services if I make it not nullable. should i got ahead and enabled it everywhere?

Reply:
The error:

Reply:
Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Microsoft.AspNetCore.Authorization.IAuthorizationService Lifetime: Transient ImplementationType: Microsoft.AspNetCore.Authorization.DefaultAuthorizationService': Unable to resolve service for type 'HealthGateway.Common.Services.IPatientService' while attempting to activate

Reply:
So the handler is used by all projects, but only some actually reach this code? If that is the case I think throwing an exception will be better than just logging an error since that would imply the startup was not properly set.
Is it possible to have a custom handler for the project? it can inherit from the base one

Reply:
Yes only Observation/Lab will check for delegation so far, I think we plan to do this for other projects in the future, I can look into making a new custom handler

Reply:
New handler implemented.